File: appsets-android-native/appsets/src/main/kotlin/xcj/app/appsets/db/room/entity/UserRelation.kt
package xcj.app.appsets.db.room.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @param id, uid or groupId
 * @param type user or group
 * @param isRelate 1 related, 0 unRelated
 */
@Entity
data class UserRelation(
    @PrimaryKey(autoGenerate = false) val id: String,
    val type: String,
    val isRelate: Int
)

File: AppSets-iOS/AppSets/com/app/appsets/db/realm/entity/UserRelation.swift
//
//  UserRelation.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/13.
//

import Foundation
import RealmSwift

class UserRelation : Object, Identifiable {
    
    @Persisted(primaryKey: true)
    var id: String
    @Persisted
    var type: String
    @Persisted
    var isRelate: Int
    
    public static let TYPE_GROUP = "group"
    
    public static let TYPE_USER = "user"
}
